sycl/include/sycl/types.hpp

```diff
@@ -56,7 +42,9 @@
 #include <sycl/marray.hpp>
 #include <sycl/multi_ptr.hpp>
 
-#ifndef __SYCL_USE_EXT_VECTOR_TYPE__
+// When ext_vector_type is not available, we rely on cl_* types from CL/cl.h
+// to represent vec storage.
+#if !defined(__HAS_EXT_VECTOR_TYPE__) || !defined(__SYCL_DEVICE_ONLY__)
 #include <sycl/detail/cl.h>
```

Comment:
Where do we use cl_* types?

Reply:
We need both a DataType and a VectorDataType.  The std::array on the host, now that we are not using ext_vector_type, needs a VectorDataType, so we use the cl_ types for that access. We need something that the math built-ins support already.  Does that make sense? I didn't test this specific inclusion though - it was there before. Maybe it's not necessary.